### PR TITLE
Adds `EndsWithNode` and `ContainsNode` typings

### PR DESCRIPTION
**Pull request by Page-**
Change-type: minor

### CODE REVIEW

File: src/AbstractSQLOptimiser.ts [hunk 1]
```diff
@@ -1552,17 +1564,8 @@ const typeRules = {
...
-		Helper<MatchFn<LikeNode>>(
-			([haystack, needle]: [TextTypeNodes, TextTypeNodes]) => [
-				'Like',
-				haystack,
-				['Concatenate', ['EmbeddedText', '%'], ['EscapeForLike', needle]],
-			],
-		),
-	),
+	Endswith: EndsWithMatcher,
+	EndsWith: EndsWithMatcher,
```
Comment (otaviojacobi) [inline src/AbstractSQLOptimiser.ts:1568]: @Page- this is duplicated, I think you want to remove one

File: src/AbstractSQLOptimiser.ts [hunk 2]
```diff
@@ -531,6 +531,18 @@ const SubtractDateMatcher = tryMatches<
 	),
 );
 
+const EndsWithMatcher = rewriteMatch(
```
Comment (otaviojacobi) [inline src/AbstractSQLOptimiser.ts:534]: I don't see why you need to extract this piece if it is not used anywhere else :thinking:

File: src/AbstractSQLOptimiser.ts (hunk 1 above)
Comment (Page-) [reply src/AbstractSQLOptimiser.ts:1568]: This is to support both `['Endswith', ...]` and `['EndsWith', ...]`

File: src/AbstractSQLOptimiser.ts (hunk 2 above)
Comment (Page-) [reply src/AbstractSQLOptimiser.ts:534]: It's used in 2 places and I didn't want to copy/paste it in both

File: src/AbstractSQLOptimiser.ts (hunk 1 above)
Comment (otaviojacobi) [reply src/AbstractSQLOptimiser.ts:1568]: :facepalm: thanks

**Review by otaviojacobi — APPROVED** [review]

**Pull request merged**